src/test/ci/team_scoreboard.test.ts

```diff
@@ -0,0 +1,306 @@
+import Team from "../../structures/team";
+import TeamScoreboard from "../../structures/team_scoreboard";
+import Player from "../../structures/player";
+
+const FIRST_TEAM_NAME = "kmq team";
+const SECOND_TEAM_NAME = "not kmqer";
+
+const FIRST_USERID = "12345";
+const TWO_USER_IDS = [FIRST_USERID, "23456"];
+const FOUR_USER_IDS = [FIRST_USERID, "23456", "252525", "1000000"];
+
+const USER_TAG = "unused";
+const AVATAR_URL = "avatarurl";
+
+let scoreboard: TeamScoreboard;
+let firstTeam: Team;
+
+/* eslint-disable prefer-arrow-callback */
+beforeEach(function () {
+    scoreboard = new TeamScoreboard();
+    firstTeam = scoreboard.addTeam(FIRST_TEAM_NAME, new Player("user#0101", FIRST_USERID, AVATAR_URL, 0));
+});
+
+describe("add a team", function () {
+    it("should add the team to the scoreboard", function () {
+        const player = new Player("second_user#1010", TWO_USER_IDS[1], AVATAR_URL, 0);
+
+        assert.strictEqual(scoreboard.hasTeam(SECOND_TEAM_NAME), false);
+        const secondTeam = scoreboard.addTeam(SECOND_TEAM_NAME, player);
+        assert.deepStrictEqual(scoreboard.getTeam(secondTeam.getName()), secondTeam);
+        assert.strictEqual(scoreboard.hasTeam(secondTeam.getName()), true);
+        assert.deepStrictEqual(Object.values(scoreboard.getTeams()), [firstTeam, secondTeam]);
+    });
+});
+
+describe("get team of player", function () {
+    it("should get the team that corresponds to the player", function () {
+        assert.deepStrictEqual(scoreboard.getTeamOfPlayer(FOUR_USER_IDS[0]), firstTeam);
+        assert.strictEqual(scoreboard.getTeamOfPlayer(FOUR_USER_IDS[1]), null);
+
+        const player = new Player("second_user#1010", FOUR_USER_IDS[1], AVATAR_URL, 0);
+        const secondTeam = scoreboard.addTeam(SECOND_TEAM_NAME, player);
+        assert.deepStrictEqual(scoreboard.getTeamOfPlayer(player.getId()), secondTeam);
+
+        const anotherPlayer = new Player(USER_TAG, FOUR_USER_IDS[2], AVATAR_URL, 0);
+        scoreboard.addPlayer(SECOND_TEAM_NAME, anotherPlayer);
+        assert.deepStrictEqual(scoreboard.getTeamOfPlayer(anotherPlayer.getId()), secondTeam);
+    });
+});
+
+describe("team deletion", function () {
+    it("should delete a team when it has no players in it", function () {
+        const player = new Player("second_user#1010", FOUR_USER_IDS[1], AVATAR_URL, 0);
+        const secondTeam = scoreboard.addTeam(SECOND_TEAM_NAME, player);
+        const anotherPlayer = new Player(USER_TAG, FOUR_USER_IDS[2], AVATAR_URL, 0);
+        scoreboard.addPlayer(SECOND_TEAM_NAME, anotherPlayer);
+        const bestPlayer = new Player(USER_TAG, FOUR_USER_IDS[3], AVATAR_URL, 0);
+        scoreboard.addPlayer(FIRST_TEAM_NAME, bestPlayer);
+        scoreboard.removePlayer(bestPlayer.getId());
+        scoreboard.removePlayer(player.getId());
+        scoreboard.removePlayer(FIRST_USERID);
+        assert.strictEqual(scoreboard.getTeam(FIRST_USERID), null);
```

Comment:
This test could also pass if first user was removed, but the team still exists. The line afterwards is comprehensive enough